lib/kitchen/driver/openstack.rb

```diff
@@ -372,29 +368,6 @@ def hints_path
         Ohai::Config[:hints_path][0]
       end
 
-      def setup_ssh(server, state)
-        if config[:key_name]
-          info "Using OpenStack keypair <#{config[:key_name]}>"
-        end
-        info "Using public SSH key <#{config[:public_key_path]}>"
-        info "Using private SSH key <#{config[:private_key_path]}>"
-        state[:ssh_key] = config[:private_key_path]
-        do_ssh_setup(state, config, server) unless config[:key_name]
-      end
-
-      def do_ssh_setup(state, config, server)
```

Comment:
Honestly, I think it's still fine to leave this out. If the user supplies a username/password in the `transport` section of the .kitchen.yml, then it will use that every time, and there's no real need to push an SSH key up there to facilitate any TK-related stuff. 

However, if there's an openstack-specific reason to have the key up there, then the transport plugin isn't going to do that for you. :)